libvast/src/system/source.cpp

```diff
@@ -172,9 +172,9 @@ source(caf::stateful_actor<source_state>* self, format::reader_ptr reader,
     if (self->state.mgr) {
       self->state.mgr->shutdown();
       self->state.mgr->out().push(detail::framed<table_slice>::make_eof());
-      self->state.mgr->out().force_emit_batches();
-      self->state.mgr->out().close();
       self->state.mgr->out().fan_out_flush();
+      self->state.mgr->out().close();
+      self->state.mgr->out().force_emit_batches();
```

Comment:
This may need to be copied over to the `datagram_source`, I think.

Reply:
The datagram source doesn't seem to manually `close()` it's connection at all, which seems to avoid all these issues. (it can get away with that because its not used in integration tests)